### ISSUE TITLE
How I am supposed to know if the image is 180 degrees rotated using OBB?

### ISSUE THREAD

**Issue by mvcastellari** [ultralytics/ultralytics#7748]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm using the pre-trained model for obb: yolov8m-obb.pt. And training it with the label format:
`class_id x1 y1 x2 y2 x3 y3 x4 y4`

The results, including the rotated angle are OK, but the problem that I am facing is that because of the 90 degree angle limitation sometimes the result will be a 180 degree rotated image. Is there any way to know if the image is 0 or 180 degrees?

My training params:
`model = YOLO("yolov8m-obb.pt")
model.train(data=myYaml.yaml, imgsz=800, batch=16, epochs=100, max_det=2, fliplr=0.0, degrees=90, box=0.1, mosaic=0, close_mosaic=0)`

Sample of image for training:

![4f52bd9c-d454-42a9-a365-4e2e4d4b40bc](https://github.com/ultralytics/ultralytics/assets/40399310/bbe7968d-e1a1-49fa-bb82-0f810de7f165)

Where points 1,2,3,4 = x1y1,x2y2,x3y3,x4y4

Just to be more clear, those images when predicted with the model will result the exactly same params, but when applying the results to rotate them, one of them will be 180 rotated:

![444](https://github.com/ultralytics/ultralytics/assets/40399310/cb79abb7-55f2-42b5-b4d9-1d6a27f2294f)

![444rot](https://github.com/ultralytics/ultralytics/assets/40399310/f511fb7c-13b2-408c-9994-edb6ae014a21)

**Comment by glenn-jocher** [comment]
@mvcastellari detecting if an image is rotated 180 degrees using an Oriented Bounding Box (OBB) can be challenging because the OBB representation is ambiguous for 180-degree rotations. The model will predict the bounding box with the least rotation necessary to encompass the object, which is limited to a 90-degree range to avoid ambiguity.

To determine if an image is 0 or 180 degrees rotated, you might consider additional post-processing steps after prediction. For instance, you could analyze the orientation of objects within the image that have a known orientation (like text or symbols) or use metadata if available.

Remember that the model's predictions are based on the data it was trained on. If the training data includes examples of objects at 0 and 180 degrees, the model might learn to differentiate between them if there are distinguishable features. However, without such features, the task becomes inherently difficult for the model.

For more detailed guidance on training parameters and model usage, please refer to our documentation at https://docs.ultralytics.com.
